src/drivers/imu/mpu9250/mpu9250.h

```diff
@@ -193,6 +200,135 @@
+#define ICMREG_20948_ACCEL_XOUT_H			(0x2D | BANK0)
+#define ICMREG_20948_INT_ENABLE_2			(0x12 | BANK0)
+#define ICMREG_20948_INT_ENABLE_3			(0x13 | BANK0)
+#define ICMREG_20948_EXT_SLV_SENS_DATA_00	(0x3B | BANK0)
+#define ICMREG_20948_GYRO_SMPLRT_DIV		(0x00 | BANK2)
+#define ICMREG_20948_GYRO_CONFIG_1			(0x01 | BANK2)
+#define ICMREG_20948_GYRO_CONFIG_2			(0x02 | BANK2)
+#define ICMREG_20948_ACCEL_SMPLRT_DIV_1		(0x10 | BANK2)
+#define ICMREG_20948_ACCEL_SMPLRT_DIV_2		(0x11 | BANK2)
+#define ICMREG_20948_ACCEL_CONFIG			(0x14 | BANK2)
+#define ICMREG_20948_ACCEL_CONFIG_2			(0x15 | BANK2)
+#define ICMREG_20948_I2C_MST_CTRL			(0x01 | BANK3)
+#define ICMREG_20948_I2C_SLV0_ADDR			(0x03 | BANK3)
+#define ICMREG_20948_I2C_SLV0_REG			(0x04 | BANK3)
+#define ICMREG_20948_I2C_SLV0_CTRL			(0x05 | BANK3)
+#define ICMREG_20948_I2C_SLV0_DO			(0x06 | BANK3)
+
+
+
+/*
+* ICM20948 register bits
+* Most of the regiser set values from MPU9250 have the same
+* meaning on ICM20948. The exceptions and values not already
+* defined for MPU9250 are defined below
+*/
+#define ICM_BIT_PWR_MGMT_1_ENABLE       	0x00
+#define ICM_BIT_USER_CTRL_I2C_MST_DISABLE   0x00
+
+#define ICM_BITS_GYRO_DLPF_CFG_197HZ		0x01
+#define ICM_BITS_GYRO_DLPF_CFG_151HZ		0x09
+#define ICM_BITS_GYRO_DLPF_CFG_119HZ		0x11
+#define ICM_BITS_GYRO_DLPF_CFG_51HZ			0x19
+#define ICM_BITS_GYRO_DLPF_CFG_23HZ			0x21
+#define ICM_BITS_GYRO_DLPF_CFG_11HZ			0x29
+#define ICM_BITS_GYRO_DLPF_CFG_5HZ			0x31
+#define ICM_BITS_GYRO_DLPF_CFG_361HZ		0x39
+#define ICM_BITS_GYRO_DLPF_CFG_MASK			0x39
+
+#define ICM_BITS_GYRO_FS_SEL_250DPS			0x00
+#define ICM_BITS_GYRO_FS_SEL_500DPS			0x02
+#define ICM_BITS_GYRO_FS_SEL_1000DPS		0x04
+#define ICM_BITS_GYRO_FS_SEL_2000DPS		0x06
+#define ICM_BITS_GYRO_FS_SEL_MASK			0x06
+
+#define ICM_BITS_ACCEL_DLPF_CFG_246HZ		0x09
+#define ICM_BITS_ACCEL_DLPF_CFG_111HZ		0x11
+#define ICM_BITS_ACCEL_DLPF_CFG_50HZ		0x19
+#define ICM_BITS_ACCEL_DLPF_CFG_23HZ		0x21
+#define ICM_BITS_ACCEL_DLPF_CFG_11HZ		0x29
+#define ICM_BITS_ACCEL_DLPF_CFG_5HZ			0x31
+#define ICM_BITS_ACCEL_DLPF_CFG_473HZ		0x39
+#define ICM_BITS_ACCEL_DLPF_CFG_MASK		0x39
+
+#define ICM_BITS_ACCEL_FS_SEL_250DPS		0x00
+#define ICM_BITS_ACCEL_FS_SEL_500DPS		0x02
+#define ICM_BITS_ACCEL_FS_SEL_1000DPS		0x04
+#define ICM_BITS_ACCEL_FS_SEL_2000DPS		0x06
+#define ICM_BITS_ACCEL_FS_SEL_MASK			0x06
+
+#define ICM_BITS_DEC3_CFG_4					0x00
+#define ICM_BITS_DEC3_CFG_8					0x01
+#define ICM_BITS_DEC3_CFG_16				0x10
+#define ICM_BITS_DEC3_CFG_32				0x11
+#define ICM_BITS_DEC3_CFG_MASK				0x11
+
+#define ICM_BITS_I2C_MST_CLOCK_370KHZ    	0x00
+#define ICM_BITS_I2C_MST_CLOCK_400HZ    	0x07	// recommended by datasheet for 400kHz target clock
+
+
+
+#define MPU_OR_ICM(m,i)					((_device_type==MPU_DEVICE_TYPE_MPU9250) ? m : i)
+
+
+#pragma pack(push, 1)
+/**
+ * Report conversation within the mpu, including command byte and
+ * interrupt status.
+ */
+struct ICMReport {
+	uint8_t		accel_x[2];
```

Comment:
MPU9250_I2C::read uses the MPUReport structure therefore this could cause issues in sending the ICMReport structure. I think we at least need a uint8_t cmd to receive something while we transfer the cmd byte in the i2c transfer. I will also further investigate this.